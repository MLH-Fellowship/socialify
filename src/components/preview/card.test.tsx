/* eslint-disable jest/no-conditional-expect */
import { render } from '@testing-library/react'

import Card from './card'

import Configuration, {
  Font,
  Pattern,
  Theme
} from '../../../common/types/configType'

test('Card #1 renders', () => {
  const config: Configuration = {
    font: Font.inter,
    logo: '',
    name: {
      value: 'project_name',
      state: true
    },
    pattern: Pattern.brickWall,
    theme: Theme.light
  }

  const { container } = render(<Card {...config} />)
  const card = container.firstElementChild!

  expect(card).toMatchSnapshot()
  expect(card.classList.contains('card-svg-wrapper')).toBe(true)

  const cardWrapper = card.querySelector<HTMLElement>('.card-wrapper')!
  expect(cardWrapper).toBeTruthy()
  expect(cardWrapper.classList.contains('card-wrapper')).toBe(true)
  expect(cardWrapper.style.fontFamily).toStrictEqual(config.font)
  expect(
    cardWrapper.classList.contains(`theme-${config.theme.toLowerCase()}`)
  ).toBe(true)
  expect(cardWrapper.querySelectorAll('.card-logo-wrapper i').length).toBe(1)
  expect(
    cardWrapper
      .querySelectorAll('.card-logo-wrapper i')?.[0]
      ?.classList.contains('devicon-github-original')
  ).toBe(true)
  expect(cardWrapper.querySelector('.card-logo-wrapper img')).toBeFalsy()
  expect(
    cardWrapper
      .querySelectorAll('.card-logo-wrapper i')[0]
      ?.classList.contains('colored')
  ).toBe(true)
  expect(cardWrapper.querySelectorAll('.card-logo-divider').length).toBe(0)
  expect(
    cardWrapper.querySelector('.card-name-name')?.textContent
  ).toStrictEqual(config.name?.value)
  expect(cardWrapper.querySelector('.card-description-wrapper')).toBeFalsy()
  expect(cardWrapper.querySelectorAll('.card-badges-wrapper').length).toBe(0)
})

test('Card #2 renders', () => {
  const config: Configuration = {
    font: Font.koho,
    logo: 'data:image/gif;base64,R0lGODlhAQABAAAAACw=',
    name: {
      value: 'project_name',
      state: true
    },
    pattern: Pattern.brickWall,
    theme: Theme.dark,
    description: {
      value: 'TEST DESCRIPTION',
      state: true
    },
    owner: {
      value: 'owner',
      state: true
    },
    language: {
      value: 'JavaScript',
      state: true
    },
    stargazers: {
      value: 1,
      state: true
    },
    forks: {
      value: 2,
      state: true
    },
    issues: {
      value: 3,
      state: true
    },
    pulls: {
      value: 4,
      state: true
    }
  }

  const { container } = render(<Card {...config} />)
  const card = container.firstElementChild!
  expect(card).toMatchSnapshot()
  expect(card.classList.contains('card-svg-wrapper')).toBe(true)

  const cardWrapper = card.querySelector<HTMLElement>('.card-wrapper')!
  expect(cardWrapper).toBeTruthy()
  expect(cardWrapper.classList.contains('card-wrapper')).toBe(true)
  expect(cardWrapper.style.fontFamily).toStrictEqual(config.font)
  expect(
    cardWrapper.classList.contains(`theme-${config.theme.toLowerCase()}`)
  ).toBe(true)
  expect(
    cardWrapper.querySelector('.card-name-name')?.textContent
  ).toStrictEqual(config.name?.value)
  expect(cardWrapper.querySelectorAll('.card-logo-wrapper img').length).toBe(1)
  expect(
    cardWrapper.querySelector<HTMLImageElement>('.card-logo-wrapper img')?.src
  ).toBe(config.logo)
  expect(cardWrapper.querySelectorAll('.card-logo-wrapper i').length).toBe(1)
  expect(cardWrapper.querySelectorAll('.card-logo-divider').length).toBe(1)
  expect(
    cardWrapper.querySelector('.card-description-wrapper')?.textContent
  ).toStrictEqual(config.description?.value)
  expect(cardWrapper.querySelectorAll('.card-badges-wrapper').length).toBe(1)
  expect(cardWrapper.querySelectorAll('.card-badges-wrapper > *').length).toBe(
    4
  )
  expect(
    cardWrapper.querySelectorAll('.card-badges-wrapper > *')[0]
      ?.firstElementChild?.textContent
  ).toStrictEqual('stars')
  expect(
    cardWrapper.querySelectorAll('.card-badges-wrapper > *')[0]
      ?.lastElementChild?.textContent
  ).toStrictEqual(`${config.stargazers?.value}`)
  expect(
    cardWrapper.querySelectorAll('.card-badges-wrapper > *')[1]
      ?.firstElementChild?.textContent
  ).toStrictEqual('forks')
  expect(
    cardWrapper.querySelectorAll('.card-badges-wrapper > *')[1]
      ?.lastElementChild?.textContent
  ).toStrictEqual(`${config.forks?.value}`)
  expect(
    cardWrapper.querySelectorAll('.card-badges-wrapper > *')[2]
      ?.firstElementChild?.textContent
  ).toStrictEqual('issues')
  expect(
    cardWrapper.querySelectorAll('.card-badges-wrapper > *')[2]
      ?.lastElementChild?.textContent
  ).toStrictEqual(`${config.issues?.value}`)
  expect(
    cardWrapper.querySelectorAll('.card-badges-wrapper > *')[3]
      ?.firstElementChild?.textContent
  ).toStrictEqual('pulls')
  expect(
    cardWrapper.querySelectorAll('.card-badges-wrapper > *')[3]
      ?.lastElementChild?.textContent
  ).toStrictEqual(`${config.pulls?.value}`)
})
