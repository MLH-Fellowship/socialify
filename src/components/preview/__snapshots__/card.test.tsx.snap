// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card #1 renders 1`] = `
<svg
  class="card-svg-wrapper"
  height="640px"
  viewBox="0 0 640 320"
  width="1280px"
  xmlns="http://www.w3.org/2000/svg"
>
  <foreignobject
    height="320"
    width="640"
    x="0"
    y="0"
  >
    <div
      class="jsx-95c052ed20ab27ca card-wrapper theme-light"
      style="font-family: Inter; background: rgb(255, 255, 255) url(data:image/svg+xml,%3Csvg%20width%3D%2242%22%20height%3D%2244%22%20viewBox%3D%220%200%2042%2044%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20d%3D%22M0%200h42v44H0V0zm1%201h40v20H1V1zM0%2023h20v20H0V23zm22%200h20v20H22V23z%22%20fill%3D%22%23eaeaea%22%20fill-opacity%3D%220.6%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E);"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <div
        class="jsx-95c052ed20ab27ca card-logo-wrapper"
      >
        <i
          class="jsx-95c052ed20ab27ca devicon-github-original colored"
        />
      </div>
      <p
        class="jsx-95c052ed20ab27ca card-name-wrapper"
        style="font-size: 40px;"
      >
        <span
          class="jsx-95c052ed20ab27ca card-name-owner"
        />
        <span
          class="jsx-95c052ed20ab27ca card-name-name"
        >
          project_name
        </span>
      </p>
    </div>
  </foreignobject>
</svg>
`;

exports[`Card #2 renders 1`] = `
<svg
  class="card-svg-wrapper"
  height="640px"
  viewBox="0 0 640 320"
  width="1280px"
  xmlns="http://www.w3.org/2000/svg"
>
  <foreignobject
    height="320"
    width="640"
    x="0"
    y="0"
  >
    <div
      class="jsx-95c052ed20ab27ca card-wrapper theme-dark"
      style="font-family: KoHo; background: rgb(0, 0, 0) url(data:image/svg+xml,%3Csvg%20width%3D%2242%22%20height%3D%2244%22%20viewBox%3D%220%200%2042%2044%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20d%3D%22M0%200h42v44H0V0zm1%201h40v20H1V1zM0%2023h20v20H0V23zm22%200h20v20H22V23z%22%20fill%3D%22%23eaeaea%22%20fill-opacity%3D%220.2%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E);"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <div
        class="jsx-95c052ed20ab27ca card-logo-wrapper"
      >
        <img
          alt="Custom logo"
          class="jsx-95c052ed20ab27ca"
          src="data:image/gif;base64,R0lGODlhAQABAAAAACw="
        />
        <span
          class="jsx-95c052ed20ab27ca card-logo-divider"
        >
          +
        </span>
        <i
          class="jsx-95c052ed20ab27ca devicon-javascript-plain "
        />
      </div>
      <p
        class="jsx-95c052ed20ab27ca card-name-wrapper"
        style="font-size: 40px;"
      >
        <span
          class="jsx-95c052ed20ab27ca card-name-owner"
        >
          owner/
        </span>
        <span
          class="jsx-95c052ed20ab27ca card-name-name"
        >
          project_name
        </span>
      </p>
      <p
        class="jsx-95c052ed20ab27ca card-description-wrapper"
      >
        TEST DESCRIPTION
      </p>
      <div
        class="jsx-95c052ed20ab27ca card-badges-wrapper"
      >
        <div
          class="jsx-b201194f83df212a badge-wrapper"
        >
          <p
            class="jsx-b201194f83df212a badge-label"
          >
            stars
          </p>
          <p
            class="jsx-b201194f83df212a badge-value"
            style="background-color: rgb(223, 179, 23);"
          >
            1
          </p>
        </div>
        <div
          class="jsx-b201194f83df212a badge-wrapper"
        >
          <p
            class="jsx-b201194f83df212a badge-label"
          >
            forks
          </p>
          <p
            class="jsx-b201194f83df212a badge-value"
            style="background-color: rgb(151, 202, 0);"
          >
            2
          </p>
        </div>
        <div
          class="jsx-b201194f83df212a badge-wrapper"
        >
          <p
            class="jsx-b201194f83df212a badge-label"
          >
            issues
          </p>
          <p
            class="jsx-b201194f83df212a badge-value"
            style="background-color: rgb(0, 126, 198);"
          >
            3
          </p>
        </div>
        <div
          class="jsx-b201194f83df212a badge-wrapper"
        >
          <p
            class="jsx-b201194f83df212a badge-label"
          >
            pulls
          </p>
          <p
            class="jsx-b201194f83df212a badge-value"
            style="background-color: rgb(254, 125, 55);"
          >
            4
          </p>
        </div>
      </div>
    </div>
  </foreignobject>
</svg>
`;
